src/group_theory/submonoid/operations.lean

```diff
@@ -101,6 +101,84 @@ def submonoid.add_submonoid_equiv (M : Type*) [mul_one_class M] :
   left_inv := λ x, by cases x; refl,
   right_inv := λ x, by cases x; refl }
 
```

Comment:
Can you instead promote `add_submonoid_equiv` on the line above to a `≃o`, passing `iff.rfl` as the proof?

Reply:
Wow this `iff.rfl` is pure black magic :D Of course it works because `to_add_submonoid` doesn't do anything, but then one may ask why we need it at all.